extensions/stackdriver/stackdriver.cc

```diff
@@ -475,6 +518,17 @@ StackdriverRootContext* StackdriverContext::getRootContext() {
   return dynamic_cast<StackdriverRootContext*>(root);
 }
 
+void StackdriverRootContext::logTCPOpen(
+    StackdriverRootContext::TcpRecordInfo& record_info,
+    const ::Wasm::Common::FlatNode& peer_node) {
+  record_info.request_info->tcp_connection_state =
```

Comment:
nit: Seems weird to me to make this a function. It handles a very specific case, instead of a general functionality. How about just remove this function and copy the function body into that if block at line 437?

Reply:
moved it for now.. but will refactor RecordTcp function in a different PR later.. feel it's become too big..